lib/ansible/modules/cloud/vmware/vmware_resource_pool.py

```diff
@@ -0,0 +1,326 @@
+        cpu_alloc_shares = vim.SharesInfo()
+        cpu_alloc_shares.level = self.cpu_shares
+        cpu_alloc.shares =  cpu_alloc_shares
+        rp_spec.cpuAllocation=cpu_alloc
+        mem_alloc=vim.ResourceAllocationInfo()
+        mem_alloc.limit=int(self.mem_limit)
+        mem_alloc.expandableReservation=self.mem_expandable_reservations
+        mem_alloc.reservation=int(self.mem_reservation)
+        mem_alloc_shares = vim.SharesInfo()
+        mem_alloc_shares.level = self.mem_shares
+        mem_alloc.shares =  mem_alloc_shares
+        rp_spec.memoryAllocation=mem_alloc
+
+        self.dc_obj = find_datacenter_by_name(self.content, self.datacenter)
+        self.cluster_obj = find_cluster_by_name_datacenter(self.dc_obj, self.cluster)
+        rootResourcePool = self.cluster_obj.resourcePool
+        task = rootResourcePool.CreateResourcePool(self.resource_pool, rp_spec)
+        #wait_for_task(task)
+
+        self.module.exit_json(changed=changed)
+
+    def check_rp_state(self):
+
+        self.host_obj, self.cluster_obj = self.find_host_by_cluster_datacenter()
+        self.resource_pool_obj = self.select_resource_pool(self.host_obj)
+
+        if self.resource_pool_obj is None:
+            return 'absent'
+        else:
+            return 'present'
+
+
+def main():
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(dict(datacenter=dict(required=True, type='str'),
+                              cluster=dict(required=True, type='str'),
+                              resource_pool=dict(required=True, type='str'),
+                              hostname=dict(required=True, type='str'),
+                              username=dict(required=True, type='str'),
+                              password=dict(required=True, type='str', no_log=True),
+                              mem_shares=dict(type='str', default="normal", choices=['high','custom','normal', 'low']),
+                              mem_limit=dict(type='int',default="-1"),
+                              mem_reservation=dict(type='int',default="0"),
+                              mem_expandable_reservations=dict(type='bool',default="True"),
+                              cpu_shares=dict(type='str', default="normal", choices=['high','custom','normal', 'low']),
+                              cpu_limit=dict(type='int',default="-1"),
+                              cpu_reservation=dict(type='int',default="0"),
+                              cpu_expandable_reservations=dict(type='bool',default="True"),
+                              state=dict(default='present', choices=['present', 'absent'], type='str')))
+
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
+
+    if not HAS_PYVMOMI:
+        module.fail_json(msg='pyvmomi is required for this module')
+
+    vmware_rp = VMwareResourcePool(module)
+    vmware_rp.process_state()
+
+from ansible.module_utils.vmware import *
+from ansible.module_utils.basic import *
```

Comment:
Hehe,
We will be getting this enforced by CI at somepoint as well, so it's good to get in the habit now :)